week7/src/components/LetterForm.tsx

```diff
@@ -0,0 +1,160 @@
+import styled from 'styled-components';
+import { useRef } from 'react';
+import { Letter } from '../types/Letter';
+import { useNavigate } from 'react-router-dom';
+import { client } from '../services';
+
+interface LetterFormProps {
+  letterInfo: Letter;
+}
+
+interface BodyObject {
+  [key: string]: string; //index signature
```

Comment:
개인적으로 해당 타입지정 선호하지 않아,,
타입 지정을 하는 의미가 조금 옅어지는 느낌이고, 단지 에러를 피하기 위한 방법이라고 생각해서!

```suggestion
  name: string;
  content: string;
  password: string;
  hint: string;
  images?: string;
```

어느 프로퍼티가 있을지 아니까, 나라면 위처럼 선언했을 것 같아!

Reply:
아 글쿤아... 맞아 나도 단지 에러를 피하고 싶었어
참고해서 다시 해볼게 고마워